### DISCUSSION TITLE
How to enable VAAPI with intel NUC i5-1135G7 and how to enable 4k recording

### DISCUSSION THREAD

**Discussion by xliang76** [roflcoopter/viseron#868] [Q&A]
I run ubuntu 22.04 in this NUC which is based on i5-1135G7. And I used the roflcoopter/amd64-viseron:latest and roflcoopter/viseron:latest. But I can't enable the hardware acceleration. When it start, it always say
`****** Checking for hardware acceleration platforms ******
OpenCL is available!
VA-API cannot be used
CUDA cannot be used`

My config.yaml like this:
`ffmpeg:
  camera:
    camera_1:  # This value has to be unique across all cameras
      name: test se
      host: !secret main_ip
      port: !secret port
      path: !secret path1
      username: !secret username
      password: !secret password
      width: 3840
      height: 2160
      fps: 15
      codec: hevc
\#      hwaccel_args:
\#          - -hwaccel
\#          - vaapi
\#          - -hwaccel_device
\#          - /dev/dri/renderD128
\#          - -hwaccel_output_format
\#          - vaapi
\#      recorder:
\#        codec: hevc_vaapi
\#        hwaccel_args:
\#          - -hwaccel
\#          - vaapi
\#          - -hwaccel_device
\#          - /dev/dri/renderD128
\#          - -hwaccel_output_format
\#          - vaapi
      substream:
        port: !secret port
        path: !secret path2
        fps: 15
        codec: hevc
\#        hwaccel_args:
\#         - -hwaccel
\#         - vaapi
\#         - -hwaccel_device
\#         - /dev/dri/renderD128
\#          - -hwaccel_output_format
\#          - vaapi

\#\# Then add an object detector
darknet:
  object_detector:
    cameras:
      camera_1:  # Attach detector to the configured camera_1 above
        fps: 5
        scan_on_motion_only: false  # Scan for objects even when there is no motion
        labels:
          - label: person
            confidence: 0.75
            trigger_recorder: true

\#\# You can also use motion detection
mog2:
  motion_detector:
    cameras:
      camera_1:  # Attach detector to the configured camera_2 above
        fps: 5

\#\# To tie everything together we need to configure one more component.
nvr:
  camera_1:  # Run NVR for camera_1

\# Now you can restart Viseron and you should be good to go!`

If I turn on the hwaccel_args, I will get "libva: /usr/lib/x86_64-linux-gnu/dri/iHD_drv_video.so init failed" errors

I only have one camera which main is 4k and sub is 704*576. when I check the live view and check record, it only show me the 704*576 resolution. Why it doesn't record in 4k?

So, I have tow questions. One is how to enable the hardware acceleration. Another one is how to make it recording 4k and live view 4k.
Thank you for any help

**Comment by roflcoopter** [comment]
Can you try if the docker tag `roflcoopter/amd64-viseron:3.0.0b12` works?
It includes upgraded drivers which might help here.

Recordings should be 4k, but the live stream is currently limited to an mjpeg stream of what Viseron uses for detection.
So the recordings will be created from the main stream and the live stream is created from the substream.

Better support for live streams is planned for 3.1
